ansible_collections/arista/avd/roles/eos_designs/templates/connected_endpoints/monitor-sessions.j2

```diff
@@ -0,0 +1,67 @@
+{# Leaf connected_endpoint monitor-sessions #}
+{% if monitor_sessions_data | length > 0 %}
+monitor_sessions:
+{%     for sname, group in monitor_sessions_data | groupby("name") %}
+{%         set group_roles = group | map(attribute='role', default='notdefined') | list %}
+{%         if 'source' in group_roles and 'destination' in group_roles %}
+  - name: {{ sname }}
+    sources:
+{%             for session in group | selectattr('role', 'equalto', 'source') %}
+{%                 if session.interfaces is arista.avd.defined %}
+      - name: {{ session.interfaces | join(", ") }}
```

Comment:
this should be single interface.

Reply:
It's plural because of the comment above. It's possible to have multiple source if's for the same session in one config. And if this is the case, we can add all of the sources to one line using the join (instead of one line per interface)